app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import LoadingSVG from 'components/LoadingSVG';
+import { humanReadableDate, isNilOrEmptyString, objectQuery } from 'services/helpers';
+import { exploreConnection } from 'components/Connections/Browser/GenericBrowser/apiHelpers';
+import * as React from 'react';
+import capitalize from 'lodash/capitalize';
+import If from 'components/If';
+import FolderIcon from '@material-ui/icons/Folder';
+import makeStyle from '@material-ui/core/styles/makeStyles';
+import Table from 'components/Table';
+import TableHeader from 'components/Table/TableHeader';
+import TableRow from 'components/Table/TableRow';
+import TableCell from 'components/Table/TableCell';
+import TableBody from 'components/Table/TableBody';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { useLocation } from 'react-router-dom';
+
+const useStyle = makeStyle(() => {
+  return {
+    nameWrapper: {
+      display: 'flex',
+      gap: '5px',
+    },
+    grid: {
+      height: '100%',
+      marginTop: '15px',
+    },
+    loadingContainer: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      height: '100%',
+    },
+  };
+});
+
+export function GenericBrowser({ selectedConnection }) {
+  const loc = useLocation();
+  const queryParams = new URLSearchParams(loc.search);
+  const pathFromUrl = queryParams.get('path') || '/';
+  const [entities, setEntities] = React.useState([]);
+  const [loading, setLoading] = React.useState(true);
+  const [error, setError] = React.useState(null);
+  const [path, setPath] = React.useState(pathFromUrl);
+  const classes = useStyle();
+
+  const fetchEntities = async () => {
+    try {
+      const res = await exploreConnection({
+        connectionid: selectedConnection,
+        path,
+      });
+      setEntities(res.entities);
+      setLoading(false);
+    } catch (e) {
+      setLoading(false);
+      setError(`Failed to explore connection : "${e.message}"`);
+    }
+  };
+  const onExplore = (entityName) => {
+    if (path === '/') {
+      setPath(`/${entityName}`);
+    } else {
+      setPath(`${path}/${entityName}`);
```

Comment:
Should this use `path` on the selected entity rather than constructing a path on the client?

Reply:
The selected entity has the name. If this is one of the root folders it will have a preceeding forward slash otherwise the new path will be existing path + the entity name.

Eg:
```
- dir1
- dir2
- dr3
```
clicking on `dir1` will generate a path `/dir1`

If under `dir1` the contents are

```
- dir12
- dir34
- dir56
```
and clicking on `dir34` will generate `/dir1/dir34`. Here `dir34` is the entity name.